src/OpenTelemetry/Metrics/Export/MetricProcessor.cs

```diff
@@ -36,6 +38,11 @@ public abstract class MetricProcessor
         /// <param name="metricName">the name of the instrument.</param>
         /// <param name="labelSet">the labelSet associated with the instrument.</param>
         /// <param name="aggregator">the aggregator used.</param>
-        public abstract void Process(string meterName, string metricName, LabelSet labelSet, Aggregator<double> aggregator);        
+        public abstract void Process(string meterName, string metricName, LabelSet labelSet, Aggregator<double> aggregator);
+
+        /// <summary>
+        /// Finish the current collection cycle and return the metrics.
```

Comment:
I don't understand what this method does based on this description. Can you please make it more elaborate. Especially as it's very different from the `SpanProcessor`

Reply:
I added comments to make it clear that this is called at the end of every collection cycles.  MetricProcessor can chose to clear it state (if its stateless) at this call.
(The spec for this are not yet final, so there may be some changes later)

This is not same as ShutDownAsync for `SpanProcessor`